### ISSUE TITLE
RxJava 2 variant

### ISSUE THREAD

**Issue by vanniktech** [trello/navi#66]
If no one else is working on this, I'd be happy to port it to RxJava 2.

**Comment by dlew** [comment]
I don't think anyone is at the moment, go for it.

I'd be willing to host it here, side-by-side with the RxJava1 version.

**Comment by vanniktech** [comment]
Alright cool, do you mind already creating a `2.x` branch or whatever name you prefer? I'd open my PR against that branch once it's done.

**Comment by dlew** [comment]
Why not just on `master`?

**Comment by vanniktech** [comment]
How do you plan supporting two versions in parallel with only one branch? Do you want to add a second module?

**Comment by dlew** [comment]
Yeah. I'd want two modules anyways because I don't want to drop support for 1.x suddenly either.

**Comment by vanniktech** [comment]
Alright and then one 'common' module or just copy paste the classes?

**Comment by dlew** [comment]
My common strategy in situations like this is to start with copy/paste, then when you're done see how much we can collapse.

(Reasoning being that there may be less common code than you think and it's much easier to merge duplicated code later vs. split it up.)

**Comment by vanniktech** [comment]
Early preview can be seen here #67

**Comment by TillSimon** [comment]
FYI: RxJava 2.0.0 was just released

**Comment by dlew** [comment]
I've been working on Navi v2 which supports RxJava 2. A snapshot release is now available on the Sonatype snapshots repository: https://github.com/trello/navi/tree/2.x#installation

You can read up on what's changed (so far) here: https://github.com/trello/navi/blob/2.x/CHANGELOG.md#20-unreleased

Next steps is getting RxLifecycle to work with RxJava 2. (Navi came first because part of RxLifecycle uses it.) Then once both are ready, I'll push them out.
